x/metadata/keeper/query_server.go

```diff
@@ -73,10 +73,75 @@ func (k Keeper) GroupContext(c context.Context, req *types.GroupContextRequest)
 	return &types.GroupContextResponse{}, nil
 }
 
-// Record returns a collection of the records in a scope or a specific one by name
-func (k Keeper) Record(c context.Context, req *types.RecordRequest) (*types.RecordResponse, error) {
-	// TODO
-	return &types.RecordResponse{}, nil
+// RecordByScopeUUID returns a collection of the records in a scope or a specific one by name
+func (k Keeper) RecordByScopeUUID(c context.Context, req *types.RecordByScopeUUIDRequest) (*types.RecordByScopeUUIDResponse, error) {
```

Comment:
Here too with the plural `RecordsByScopeUUID` and `RecordsByScopeID` below.

Reply:
Agreed.